src/main/java/eu/dissco/backend/repository/DigitalMediaObjectRepository.java

```diff
@@ -1,103 +1,225 @@
 package eu.dissco.backend.repository;
 
+import static eu.dissco.backend.database.jooq.Tables.NEW_ANNOTATION;
 import static eu.dissco.backend.database.jooq.Tables.NEW_DIGITAL_MEDIA_OBJECT;
+import static eu.dissco.backend.database.jooq.Tables.NEW_DIGITAL_SPECIMEN;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import eu.dissco.backend.domain.DigitalMediaObject;
+import eu.dissco.backend.domain.JsonApiData;
 import java.util.List;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.jooq.DSLContext;
+import org.jooq.Field;
 import org.jooq.Record;
 import org.jooq.Record1;
+import org.jooq.impl.DSL;
 import org.springframework.stereotype.Repository;
 
 @Repository
 @RequiredArgsConstructor
+@Slf4j
 public class DigitalMediaObjectRepository {
 
   private final ObjectMapper mapper;
   private final DSLContext context;
 
   public DigitalMediaObject getLatestDigitalMediaById(String id) {
     return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
-        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID)
-        .from(NEW_DIGITAL_MEDIA_OBJECT)
+        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID).from(NEW_DIGITAL_MEDIA_OBJECT)
         .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
         .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc())
         .fetchOne(this::mapToMultiMediaObject);
   }
 
-  private DigitalMediaObject mapToMultiMediaObject(Record dbRecord) {
-    try {
-      return new DigitalMediaObject(
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.VERSION),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.CREATED),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.TYPE),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.FORMAT),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.SOURCE_SYSTEM_ID),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DATA).data()),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ORIGINAL_DATA).data())
-      );
-    } catch (JsonProcessingException e) {
-      throw new RuntimeException(e);
-    }
+  public JsonApiData getLatestDigitalMediaObjectByIdJsonResponse(String id) {
+
+    Field<Integer> maxSpeciesVersion = DSL.max(NEW_DIGITAL_SPECIMEN.VERSION)
+        .as("maxSpeciesVersion");
+
+    var specimenRecentVersions = context.select(NEW_DIGITAL_SPECIMEN.ID, maxSpeciesVersion)
```

Comment:
Not sure what is happening here. It feels like you first make a new table with all latest version of the digital Specimen. This is a very heavy query, and I am not sure why but I might misread it. I would say if you need the digitalMedia of the object you could just query that table. If need specimen info on it you can use the physicalSpecimenId in the digitalMedia to look up more info?